homeassistant/components/switchbot_cloud/__init__.py

```
@@ -189,3 +199,131 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+) -> Callable[[HomeAssistant, str, web.Request], Awaitable[None]]:
+    """Create a webhook handler."""
+
+    async def _internal_handle_webhook(
+        hass: HomeAssistant, webhook_id: str, request: web.Request
+    ) -> None:
+        """Handle webhook callback."""
+        data = await request.json()
+        _LOGGER.info("Receive data from webhook %s", repr(data))
+
+        if not isinstance(data, dict):
+            _LOGGER.error(
+                "Received invalid data from switchbot webhook. Data needs to be a dictionary: %s",
+                data,
+            )
+            return
+
+        if "eventType" not in data or data["eventType"] != "changeReport":
+            _LOGGER.error(
+                'Received invalid data from switchbot webhook. Attribute eventType is missing or not equals to "changeReport": %s',
+                data,
+            )
+            return
+
+        if "eventVersion" not in data or data["eventVersion"] != "1":
+            _LOGGER.error(
+                'Received invalid data from switchbot webhook. Attribute eventVersion is missing or not equals to "1": %s',
+                data,
+            )
+            return
+
+        if "context" not in data or not isinstance(data["context"], dict):
+            _LOGGER.error(
+                "Received invalid data from switchbot webhook. Attribute context is missing or not instance of dict: %s",
+                data,
+            )
+            return
+
+        if "deviceType" not in data["context"] or "deviceMac" not in data["context"]:
+            _LOGGER.error(
+                "Received invalid data from switchbot webhook. Missing deviceType or deviceMac: %s",
+                data,
+            )
+            return
+
+        deviceMac = data["context"]["deviceMac"]
+
+        if deviceMac not in coordinators_by_id:
+            _LOGGER.error(
+                "Received data for unknown entity from switchbot webhook: %s", data
+            )
+            return
```

I personally would just silently discard invalid requests

---

or maybe just log them on debug

---

okay, I'll take note

---

done